### DISCUSSION TITLE
[Feature] View photos from albums shared with me in timeline

### DISCUSSION THREAD

**Discussion by djvaldez** [immich-app/immich#1779] [Feature Request]
### The feature

On the main timeline page, a user will only see the photos they uploaded. This works similar to Google Photos, but it could be useful for users to at least have the option of seeing other users' photos shared with them on their timeline too. Is there a plan for this, or would it be infeasible based on performance/database requirements?

### Platform

- [X] Server
- [X] Web
- [X] Mobile

**Comment by alextran1502** [comment]
It is feasible, I can see some additional queries that related to getting the shared asset for the main timeline.

**Reply by Kevinjil** [reply]
Would be nice to have a per-album checkbox.

**Reply by frederikb96** [reply]
This feature could be addressed with #1587

**Reply by Phlogi** [reply]
In my opinion, "add to library" is more confusing for the users instead of just setting it simply per shared-to-me albums.

**Reply by lucianomunoz** [reply]
I would call it "Display in timeline" or "Add to timeline"

**Comment by Phlogi** [comment]
Can you replace the original description with this? I was just going to request this feature and found this duplicate. 

Introduce a checkbox option on shared-to-me albums that allows users to display assets (photos and videos) from these albums in their personal timeline.

**Implementation proposal deatils**
- Checkbox Option: Add a checkbox on each shared-to-me album labeled "Display assets in my timeline."
- Functionality: When the checkbox is checked, all assets from the selected shared album will be displayed in the user's personal timeline.
- User Interface: The checkbox will be located in the album settings. Album list, three dot menu --> settings (new).

**Benefit for user**     
This is the missing killer feature for very close people to share assets, e.g. of their family. This finally allows having one view over all relevant photos, without the need to switch to family albums when looking for assets. 

PS: This is a long awaited missing feature when reading through the existing discussions about sharing in https://github.com/immich-app/immich/discussions/7038 and https://github.com/immich-app/immich/issues/12614

**Reply by Phlogi** [reply]
A late Christmas present: I implemented the functionality and created a PR, let's hope the immich team approves it. 
You might want to leave some reactions: https://github.com/immich-app/immich/pull/24938
<img width="349" height="88" alt="image" src="https://github.com/user-attachments/assets/0cf77c25-eb65-42af-a26d-95270f4a220e" />

**Reply by DJ-Kale** [reply]
Got really excited about this until I saw the Immich team's response to the work you did... Completely disrespectful to your time, effort & energy into an open source project.

**Comment by JamesMenetrey** [comment]
I think this feature would be very useful in a family setup! As mentioned in other discussions (#3984, #7047, #10425), many users would like the ability to see shared photos in their main timeline. Currently, when family members or friends share photos, these images only appear in shared albums and never show up chronologically in the viewer's timeline, requiring users to navigate to multiple places to see all relevant photos. From what I have gathered in these duplicated discussions, many users have described family scenarios where they have separate accounts but want a unified view of important events without constantly switching between albums. Several users mentioned that older family members find the current separation particularly confusing.

As a solution, the most popular request in the duplicated discussions is for a configurable option that would allow users to choose which shared content appears in their timeline. This could be implemented as a global setting, or a per-album toggle.

**Reply by lucianomunoz** [reply]
Totally, this is the exact reason I came here, I want to share my server with my mom, dad, girlfriend but be able to chose which albums shared can go in my timeline. 

Same as with search I wish search was a bit stronger just one searchbar to search everything instead of having for instance a search bar for people under people.

The less places we need to move around to look for things the easier they're to find!

**Reply by srvprivate** [reply]
This is exactly the reason I am here; I thought it was a bug actually. Glad everyone is in agreement here. I've been working on multi-family albums for one or two accounts. I am really seeing the power of this program take off in the last year. I have some scenarios that I'd like to work if I'm going to do multi-family with different rights to what they can all see. It's doing good so far, but we need this fixed and the Shared Faces.

**Comment by carlolars** [comment]
I'd like to have an option to partner sharing that instead of adding _all_ photos from my partner to my timeline only photos in albums shared by my partner to me should be added to my timeline.
This way my timeline won't get cluttered by my wifes irrelevant photos and she can take whatever photos she wants without having to think that this will show up on my timeline.

**Reply by carlolars** [reply]
Maybe this is not a necessary a "partner" sharing feature, more of a "put all photos in albums shared with me from this user on my timeline".

**Comment by jo-gross** [comment]
I’m also interested in such a feature. I’d like to suggest another potential use case for consideration.

In addition to family members — already mentioned above — who also have photos in Immich, I also have several friends with whom I regularly want to share my albums. Until now, I’ve always shared links, but I recently created a single account with no storage space. These users should exclusively have the ability to view photos from the shared albums.

Currently, they must select at least one of their own photos to display the timeline (they only checked the iOS app). It would be very useful to be able to view the timeline without having any own photos — showing only the shared photos.

I think there are therefore the following user scenarios:
- Immich users (with their own photos on the instance)
- Immich users (without their own photos on the instance)

and of course
- Users with their own Immich instance — where cross-sharing would be incredibly cool, as already mentioned in #13621, #13620 or #728, but unfortunately not planned for now :(

**Reply by Phlogi** [reply]
I consolidate all inputs regarding sharing and created a theoretical concept here: https://github.com/immich-app/immich/blob/1bcffa6c39843706f29d256e906ffba7b423ce12/docs/docs/features/better-sharing.md

This was shared with the immich team, feel free to add comments or ideas for improvements.

**Comment by wzjoriv** [comment]
Any update?

**Reply by bo0tzz** [reply]
https://justinmayer.com/posts/any-updates/

**Comment by DJ-Kale** [comment]
+1 to this. Recently imported my photos over from Nextcloud & my only issue is the "Shared Photos" I uploaded are only showing up on my timeline despite me sharing them with my wife. I'm aware of the partner sharing option but I take a lot of junk pictures of cabling, receipts, labels etc that I'd rather not pollute her feed with.

For example, we have a Vegas trip from ~10 years ago that only shows up on the bottom of my timeline, but not hers, despite it being shared. Ideally a setting to make shared albums appear on the timeline would be very helpful.

**Comment by YarosMallorca** [comment]
This isn't how it already works?
When someone uploads assets to an album I shared via link, it also appears in my main timeline. This isn't the case when sharing albums between users?

**Reply by Phlogi** [reply]
Unfortunately not, no.

**Comment by mwillers** [comment]
Would greatly appreciate that feature (or something that works like the "shared library" feature on ios - i.e. selecting specific photos that are shared with a partner (and not the complete library)). Not having this prevents me from going all in on immich.

**Comment by mwillers** [comment]
Interestingly, it seems to work just fine in the maps feature ("Include shared albums" - which displays images from albums shared with me on the map) but is not enabled in Photos.
Not sure if there is a substantial difference between showing shared + own photos on a map vs showing them on the timeline.

**Comment by artamrj** [comment]
@alextran1502 Hi, is there any update on this feature? This is really a must-have. Could you please consider adding it to the roadmap?

**Reply by mpeter50** [reply]
@Phlogi 's fork has an implementation, you could probably use that: https://github.com/immich-app/immich/pull/24938

**Reply by Phlogi** [reply]
Bad long term idea. My PR was rejected because of the open sharing/permission issue. That one is on the roadmap.

https://github.com/immich-app/immich/pull/24938

**Reply by YarosMallorca** [reply]
https://github.com/immich-app/immich/issues/12614#issuecomment-4588031199
